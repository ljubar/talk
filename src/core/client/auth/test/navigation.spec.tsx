import { ReactTestRenderer } from "react-test-renderer";
import sinon from "sinon";

import { timeout } from "talk-common/utils";
import create from "./create";
import { settings } from "./fixtures";

async function createTestRenderer(
  initialView: string
): Promise<ReactTestRenderer> {
  const resolvers = {
    Query: {
      settings: sinon.stub().returns(settings),
    },
  };
  const windowMock = sinon.mock(window);
  windowMock.expects("resizeTo");
  const { testRenderer } = create({
    // Set this to true, to see graphql responses.
    logNetwork: false,
    resolvers,
    initLocalState: localRecord => {
      localRecord.setValue(initialView, "view");
    },
  });
  await timeout();
  windowMock.restore();
  return testRenderer;
}

it("renders sign in form", async () => {
  const testRenderer = await createTestRenderer("SIGN_IN");
  expect(testRenderer.toJSON()).toMatchSnapshot();
});

it("navigates to sign up form", async () => {
  const testRenderer = await createTestRenderer("SIGN_IN");
  testRenderer.root
    .findByProps({ id: "signIn-gotoSignUpButton" })
    .props.onClick();
  expect(testRenderer.toJSON()).toMatchSnapshot();
});

it("navigates to sign in form", async () => {
  const testRenderer = await createTestRenderer("SIGN_UP");
  testRenderer.root
    .findByProps({ id: "signUp-gotoSignInButton" })
    .props.onClick();
  expect(testRenderer.toJSON()).toMatchSnapshot();
});

it("navigates to forgot password form", async () => {
  const testRenderer = await createTestRenderer("SIGN_IN");
  testRenderer.root
    .findByProps({ id: "signIn-gotoForgotPasswordButton" })
    .props.onClick();
  expect(testRenderer.toJSON()).toMatchSnapshot();
});
