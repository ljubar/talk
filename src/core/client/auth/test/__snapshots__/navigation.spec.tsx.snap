// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigates to forgot password form 1`] = `
<div>
  <div>
    <div
      className="Flex-root Bar-root Flex-flex Flex-justifyCenter Flex-alignCenter"
    >
      <div>
        <h1
          className="Typography-root Typography-heading2 Typography-colorTextPrimary Typography-alignCenter Title-root"
        >
          Forgot Password
        </h1>
      </div>
    </div>
    <div
      className="Main-root"
    >
      <form
        autoComplete="off"
        onSubmit={[Function]}
      >
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        >
          <p
            className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
          >
            Enter your email address below and we will send you a link to
reset your password.
          </p>
          <div
            className="HorizontalGutter-root FormField-root HorizontalGutter-half"
          >
            <label
              className="Typography-root Typography-inputLabel Typography-colorTextPrimary InputLabel-root"
            >
              Email Address
            </label>
            <input
              className="TextField-root TextField-colorRegular TextField-fullWidth"
              disabled={false}
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
              value=""
            />
          </div>
          <button
            className="BaseButton-root Button-root Button-sizeLarge Button-colorPrimary Button-variantFilled Button-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            Send Email
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`navigates to sign in form 1`] = `
<div>
  <div>
    <div
      className="Flex-root Bar-root Flex-flex Flex-justifyCenter Flex-alignCenter"
    >
      <div>
        <h1
          className="Typography-root Typography-heading2 Typography-colorTextPrimary Typography-alignCenter Title-root"
        >
          Sign In
        </h1>
        <h1
          className="Typography-root Typography-heading4 Typography-colorTextPrimary Typography-alignCenter Subtitle-root"
        >
          to join the conversation
        </h1>
      </div>
    </div>
    <div
      className="Flex-root SubBar-root Flex-flex Flex-justifyCenter Flex-alignCenter"
    >
      <div>
        <div
          className="Flex-root Typography-root Typography-bodyCopy Typography-colorTextPrimary Flex-flex"
        >
          <span>
            Don't have an account? 
          </span>
          <button
            className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
            id="signIn-gotoSignUpButton"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            Sign Up
          </button>
        </div>
      </div>
    </div>
    <div
      className="Main-root"
    >
      <div
        className="HorizontalGutter-root HorizontalGutter-oneAndAHalf"
      >
        <form
          autoComplete="off"
          onSubmit={[Function]}
        >
          <div
            className="HorizontalGutter-root HorizontalGutter-full"
          >
            <div
              className="HorizontalGutter-root FormField-root HorizontalGutter-half"
            >
              <label
                className="Typography-root Typography-inputLabel Typography-colorTextPrimary InputLabel-root"
              >
                Email Address
              </label>
              <input
                className="TextField-root TextField-colorRegular TextField-fullWidth"
                disabled={false}
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
              />
            </div>
            <div
              className="HorizontalGutter-root FormField-root HorizontalGutter-half"
            >
              <label
                className="Typography-root Typography-inputLabel Typography-colorTextPrimary InputLabel-root"
              >
                Password
              </label>
              <div
                className="PasswordField-fullWidth PasswordField-root"
              >
                <div
                  className="PasswordField-wrapper"
                >
                  <input
                    className="PasswordField-colorRegular PasswordField-fullWidth PasswordField-input"
                    disabled={false}
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <div
                    className="PasswordField-icon"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Show password"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-sm"
                    >
                      visibility
                    </span>
                  </div>
                </div>
              </div>
              <div
                className="Flex-root Flex-flex Flex-justifyFlexEnd"
              >
                <button
                  className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
                  disabled={false}
                  id="signIn-gotoForgotPasswordButton"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  Forgot your password?
                </button>
              </div>
            </div>
            <button
              className="BaseButton-root Button-root Button-sizeLarge Button-colorBrand Button-variantFilled Button-fullWidth"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              type="submit"
            >
              <span
                aria-hidden="true"
                className="Icon-root ButtonIcon-root Icon-md"
              >
                email
              </span>
              <span>
                Sign in with Email
              </span>
            </button>
          </div>
        </form>
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`navigates to sign up form 1`] = `
<div>
  <div>
    <div
      className="Flex-root Bar-root Flex-flex Flex-justifyCenter Flex-alignCenter"
    >
      <div>
        <h1
          className="Typography-root Typography-heading2 Typography-colorTextPrimary Typography-alignCenter Title-root"
        >
          Sign Up
        </h1>
        <h1
          className="Typography-root Typography-heading4 Typography-colorTextPrimary Typography-alignCenter Subtitle-root"
        >
          to join the conversation
        </h1>
      </div>
    </div>
    <div
      className="Flex-root SubBar-root Flex-flex Flex-justifyCenter Flex-alignCenter"
    >
      <div>
        <div
          className="Flex-root Typography-root Typography-bodyCopy Typography-colorTextPrimary Flex-flex"
        >
          <span>
            Already have an account? 
          </span>
          <button
            className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
            id="signUp-gotoSignInButton"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            Sign In
          </button>
        </div>
      </div>
    </div>
    <div
      className="Main-root"
    >
      <div
        className="HorizontalGutter-root HorizontalGutter-oneAndAHalf"
      >
        <form
          autoComplete="off"
          onSubmit={[Function]}
        >
          <div
            className="HorizontalGutter-root HorizontalGutter-full"
          >
            <div
              className="HorizontalGutter-root FormField-root HorizontalGutter-half"
            >
              <label
                className="Typography-root Typography-inputLabel Typography-colorTextPrimary InputLabel-root"
              >
                Email Address
              </label>
              <input
                className="TextField-root TextField-colorRegular TextField-fullWidth"
                disabled={false}
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
              />
            </div>
            <div
              className="HorizontalGutter-root FormField-root HorizontalGutter-half"
            >
              <label
                className="Typography-root Typography-inputLabel Typography-colorTextPrimary InputLabel-root"
              >
                Username
              </label>
              <p
                className="Typography-root Typography-detail Typography-colorTextSecondary"
              >
                You may use “_” and “.” Spaces not permitted.
              </p>
              <input
                className="TextField-root TextField-colorRegular TextField-fullWidth"
                disabled={false}
                name="username"
                onChange={[Function]}
                placeholder="Username"
                type="text"
                value=""
              />
            </div>
            <div
              className="HorizontalGutter-root FormField-root HorizontalGutter-half"
            >
              <label
                className="Typography-root Typography-inputLabel Typography-colorTextPrimary InputLabel-root"
              >
                Password
              </label>
              <p
                className="Typography-root Typography-detail Typography-colorTextSecondary"
              >
                Must be at least ⁨8⁩ characters
              </p>
              <div
                className="PasswordField-fullWidth PasswordField-root"
              >
                <div
                  className="PasswordField-wrapper"
                >
                  <input
                    className="PasswordField-colorRegular PasswordField-fullWidth PasswordField-input"
                    disabled={false}
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <div
                    className="PasswordField-icon"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Show password"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-sm"
                    >
                      visibility
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <button
              className="BaseButton-root Button-root Button-sizeLarge Button-colorBrand Button-variantFilled Button-fullWidth"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              type="submit"
            >
              <span
                aria-hidden="true"
                className="Icon-root ButtonIcon-root Icon-md"
              >
                email
              </span>
              <span>
                Sign up with Email
              </span>
            </button>
          </div>
        </form>
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders sign in form 1`] = `
<div>
  <div>
    <div
      className="Flex-root Bar-root Flex-flex Flex-justifyCenter Flex-alignCenter"
    >
      <div>
        <h1
          className="Typography-root Typography-heading2 Typography-colorTextPrimary Typography-alignCenter Title-root"
        >
          Sign In
        </h1>
        <h1
          className="Typography-root Typography-heading4 Typography-colorTextPrimary Typography-alignCenter Subtitle-root"
        >
          to join the conversation
        </h1>
      </div>
    </div>
    <div
      className="Flex-root SubBar-root Flex-flex Flex-justifyCenter Flex-alignCenter"
    >
      <div>
        <div
          className="Flex-root Typography-root Typography-bodyCopy Typography-colorTextPrimary Flex-flex"
        >
          <span>
            Don't have an account? 
          </span>
          <button
            className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
            id="signIn-gotoSignUpButton"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            Sign Up
          </button>
        </div>
      </div>
    </div>
    <div
      className="Main-root"
    >
      <div
        className="HorizontalGutter-root HorizontalGutter-oneAndAHalf"
      >
        <form
          autoComplete="off"
          onSubmit={[Function]}
        >
          <div
            className="HorizontalGutter-root HorizontalGutter-full"
          >
            <div
              className="HorizontalGutter-root FormField-root HorizontalGutter-half"
            >
              <label
                className="Typography-root Typography-inputLabel Typography-colorTextPrimary InputLabel-root"
              >
                Email Address
              </label>
              <input
                className="TextField-root TextField-colorRegular TextField-fullWidth"
                disabled={false}
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
              />
            </div>
            <div
              className="HorizontalGutter-root FormField-root HorizontalGutter-half"
            >
              <label
                className="Typography-root Typography-inputLabel Typography-colorTextPrimary InputLabel-root"
              >
                Password
              </label>
              <div
                className="PasswordField-fullWidth PasswordField-root"
              >
                <div
                  className="PasswordField-wrapper"
                >
                  <input
                    className="PasswordField-colorRegular PasswordField-fullWidth PasswordField-input"
                    disabled={false}
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <div
                    className="PasswordField-icon"
                    onClick={[Function]}
                    role="button"
                    tabIndex={0}
                    title="Show password"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-sm"
                    >
                      visibility
                    </span>
                  </div>
                </div>
              </div>
              <div
                className="Flex-root Flex-flex Flex-justifyFlexEnd"
              >
                <button
                  className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
                  disabled={false}
                  id="signIn-gotoForgotPasswordButton"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  Forgot your password?
                </button>
              </div>
            </div>
            <button
              className="BaseButton-root Button-root Button-sizeLarge Button-colorBrand Button-variantFilled Button-fullWidth"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              type="submit"
            >
              <span
                aria-hidden="true"
                className="Icon-root ButtonIcon-root Icon-md"
              >
                email
              </span>
              <span>
                Sign in with Email
              </span>
            </button>
          </div>
        </form>
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        />
      </div>
    </div>
  </div>
</div>
`;
